crates/swc_fast_ts_strip/src/lib.rs

```diff
@@ -611,6 +664,18 @@ impl Visit for TsStrip {
     }
 }
 
+trait U8Helper {
+    fn is_utf8_char_boundary(&self) -> bool;
+}
+
+impl U8Helper for u8 {
+    #[inline]
+    fn is_utf8_char_boundary(&self) -> bool {
+        // This is bit magic equivalent to: b < 128 || b >= 192
+        (*self as i8) >= -0x40
```

Comment:
Is this copied from some legacy code, or eyeball optimized? The expression in comment compiles to exact same code with the "optimized" one: https://godbolt.org/z/KsqjM8jad

IMO, readability is more important if they are actually the same.

Reply:
Copy from std::core::num::u8